### PR TITLE
#38 select location tests

### PR DESCRIPTION
**Pull request by thomasgauvin**
* Added unit tests for InfoCardViewModel, SearchActivityViewModel and RoutesActivityViewModel

* Added a way to decouple the ViewModels from the Context by using a ViewModelBuilder which builds ViewModels with the database, so that it is mockable

### CODE REVIEW

File: app/src/test/java/com/example/concordia_campus_guide/ViewModel/InfoCardFragmentViewModelTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.example.concordia_campus_guide.ViewModel;
```
[the remark this answers is not in the document]
Comment (thomasgauvin) [reply app/src/test/java/com/example/concordia_campus_guide/ViewModel/InfoCardFragmentViewModelTest.java]: Yes, there is a way to mock the context of the application, but it seems complicated if we need the context within AppDatabase.getInstance(context)

**Comment by thomasgauvin** [comment]
image

File: app/src/main/java/com/example/concordia_campus_guide/Helper/ViewModelFactory.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private Application mApplication;
+    private AppDatabase appDb;
+
+
+    public ViewModelFactory(Application application) {
+        mApplication = application;
+        this.appDb = AppDatabase.getInstance(application.getApplicationContext());
+    }
+
+    //all view models need to be registered in the ViewModelFactory if they want to benefit from
```
Comment (ferdousara-parvin) [inline app/src/main/java/com/example/concordia_campus_guide/Helper/ViewModelFactory.java]: put a javadoc instead?

Comment (thomasgauvin) [reply app/src/main/java/com/example/concordia_campus_guide/Helper/ViewModelFactory.java]: sike

**Pull request merged**